### PR TITLE
Add Contour 1.21.1 package

### LINKED ISSUE #3104: Changes to Contour package for new namespace-scoped Role (in Contour v1.21.0)
## Feature Request

The next release of Contour (v1.21.0) will switch from using a ClusterRole to a namespace-scoped Role to configure RBAC for leader election resources (Leases and Events). This is a small security change to limit Contour's access footprint in a cluster.

As this change adds a new resource (a new Role) to Contour's base deployment configuration, some steps will need to be taken in TCE to accommodate this change (e.g. setting the namespace on the new Role correctly).

## Describe alternatives you've considered

N/A

## Additional context

See Contour change: https://github.com/projectcontour/contour/pull/4204

cc @vmware-tanzu/pkg-contour-owners @vmware-tanzu/contour-owners

### LINKED ISSUE #4605: add Contour v1.21.1 package
## Feature Request

Contour v1.21.1 has been released, we need to add packaging for it.

https://github.com/projectcontour/contour/releases/tag/v1.21.1

cc @vmware-tanzu/pkg-contour-owners

### PR DESCRIPTION
**Pull request by sunjayBhatia**
## What this PR does / why we need it
- Modifies change-namespace overlay to accommodate new RoleBinding and
      ServiceAccount pulled out of ClusterRoleBinding for namespace specific
      access to leader election resources.
- Removes base values.yaml and adds schema.yaml
- contour.configFileContents is set as an Any type passthrough YAML
- same for envoy.service.annotations, however this field is validated to
      be a key/value map, see additional unit test
- package.yaml regenerated with `make generate-openapischema-package
      PACKAGE=contour VERSION=1.21.1`

## Which issue(s) this PR fixes
Fixes: #4605
Fixes: #3104

## Describe testing done for PR
unit test added, generated YAML and applied to basic kind cluster

## Special notes for your reviewer
N/A

### CODE REVIEW

**Comment by sunjayBhatia** [comment]
Replaces #4618

Also based on package from #4804 so draft for now while that is reviewed/merged

File: addons/packages/contour/1.21.1/package.yaml [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  version: 1.21.1
+  releaseNotes: contour 1.21.1 https://github.com/projectcontour/contour/releases/tag/v1.21.1
+  releasedAt: "2022-06-14T00:00:00Z"
+  licenses:
+  - Apache 2.0
+  template:
+    spec:
+      fetch:
+      - imgpkgBundle:
+          image: projects.registry.vmware.com/tce/contour@sha256:22604579517580e494d2e73b440c0ac197d7eda378b6021ee145f63ecbf0ea2f
```
Comment (sunjayBhatia) [inline addons/packages/contour/1.21.1/package.yaml:16]: still needs an update, will push after review

**Review by skriss — DISMISSED** [review]
LGTM!

For future reference, these are a little easier to review if you can split it in two commits: the first one just copy-pastes the previous version's package into a new directory with no changes, and then the second commit includes the updates for the new version. That way I can just review the second commit to see what changed from the previous version easily. I did something similar locally to be able to review this one.

**Comment by sunjayBhatia** [comment]
> LGTM!
> 
> For future reference, these are a little easier to review if you can split it in two commits: the first one just copy-pastes the previous version's package into a new directory with no changes, and then the second commit includes the updates for the new version. That way I can just review the second commit to see what changed from the previous version easily. I did something similar locally to be able to review this one.

ah yeah will do next time for easier reviewing 👍🏽

**Review by skriss — APPROVED** [review]

**Comment by sunjayBhatia** [comment]
https://github.com/vmware-tanzu/community-edition/pull/5061 to help fix the lint error

**Pull request merged**